pkg/reconciler/events/cloudevent/cloud_event_controller.go

```diff
@@ -73,6 +76,43 @@ func cloudEventDeliveryFromTargets(targets []string) []v1beta1.CloudEventDeliver
 	return nil
 }
 
+// EmitCloudEvents emits CloudEvents (only) for object
+func EmitCloudEvents(ctx context.Context, object runtime.Object) {
+	logger := logging.FromContext(ctx)
+	configs := config.FromContextOrDefaults(ctx)
+	sendCloudEvents := (configs.Defaults.DefaultCloudEventsSink != "")
+	if sendCloudEvents {
+		ctx = cloudevents.ContextWithTarget(ctx, configs.Defaults.DefaultCloudEventsSink)
+	}
+
+	if sendCloudEvents {
+		err := SendCloudEventWithRetries(ctx, object)
+		if err != nil {
+			logger.Warnf("Failed to emit cloud events %v", err.Error())
+		}
+	}
+}
+
+// EmitCloudEventsWhenConditionChange emits CloudEvents when there is a change in condition
+func EmitCloudEventsWhenConditionChange(ctx context.Context, beforeCondition *apis.Condition, afterCondition *apis.Condition, object runtime.Object) {
```

Comment:
I'm not sure if we should merge this function with `EmitCloudEvents`, the only difference is to check the condition.

Reply:
I think it's ok to do that in a separate PR.

Reply:
👍